core/server/lib/members/static/auth/components/Form.js

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'preact';
+
+const states = {};
+
+export default class Form extends Component {
+
+    constructor(props) {
+        super(props);
+        const includeData = props.includeData || {};
+
+        if (props.bindTo) {
+            if (states[props.bindTo]) {
+                this.state = states[props.bindTo]
+                this.state.data = { ...this.state.data, ...includeData };
+            } else {
+                states[props.bindTo] = this.state = {
+                    submitted: false,
+                    data: {...includeData}
+                }
+            }
+        } else {
+            this.state = {
+                submitted: false,
+                data: {...includeData}
+            }
+        }
+    }
+
+    wrapChildren(children, data, onInput = () => {}) {
```

Comment:
I feel `Form` in this state has got a bit too complex, with all the modifications and wrapping directly over children, it seems much harder to maintain or edit easily. Separating out components to generic works as long as they can be easily modified with just props like with rest of components, maybe this could use a refactor later or we directly use `Form` instead of its component :)

Reply:
Yeah it's deffo a bit complex atm, I thought it was nicely encapsulated though - where would you suggest moving the logic to?